### PR TITLE
[zh] Sync blog 2023-08-24-swap-beta1-graduation.md into Chinese

### PR DESCRIPTION
**Pull request by wilsonwu**
Sync blog 2023-08-24-swap-beta1-graduation.md into Chinese:

```
content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md
```

### CODE REVIEW

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+-->
+**作者**：Itamar Holder (Red Hat)
+
+**译者**：Wilson Wu (DaoCloud)
+
+<!--
+The 1.22 release [introduced Alpha support](/blog/2021/08/09/run-nodes-with-swap-alpha/) for configuring swap memory usage for Kubernetes workloads running on Linux on a per-node basis. Now, in release 1.28, support for swap on Linux nodes has graduated to Beta, along with many new improvements.
+-->
+1.22 版中[引入了 Alpha 支持](/blog/2021/08/09/run-nodes-with-swap-alpha/)，
+用于为每个在 Linux 节点上运行的 Kubernetes 工作负载配置交换内存使用情况。
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:25]:
```suggestion
Kubernetes 1.22 版本[引入了一项 Alpha 支持](/blog/2021/08/09/run-nodes-with-swap-alpha/)，
即在每个节点上为运行在 Linux 上的 Kubernetes 工作负载配置交换内存用量。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+在详细计算交换内存限制之前，有必要定义以下术语：
+
+<!--
+* `nodeTotalMemory`: The total amount of physical memory available on the node.
+* `totalPodsSwapAvailable`: The total amount of swap memory on the node that is available for use by Pods (some swap memory may be reserved for system use).
+* `containerMemoryRequest`: The container's memory request.
+-->
+* `nodeTotalMemory`：节点上可用的物理内存总量。
+* `totalPodsSwapAvailable`：节点上可供 Pod 使用的交换内存总量（某些交换内存可能会保留供系统使用）。
+* `containerMemoryRequest`: 容器的内存请求。
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:216]:
```suggestion
* `containerMemoryRequest`：容器的内存请求。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+## How do I get involved?
+-->
+## 参与其中 {#how-do-i-get-involved}
+
+<!--
+Your feedback is always welcome! SIG Node [meets regularly](https://github.com/kubernetes/community/tree/master/sig-node#meetings) and [can be reached](https://github.com/kubernetes/community/tree/master/sig-node#contact) via [Slack](https://slack.k8s.io/) (channel **#sig-node**), or the SIG's [mailing list](https://groups.google.com/forum/#!forum/kubernetes-sig-node). A Slack channel dedicated to swap is also available at **#sig-node-swap**.
+-->
+随时欢迎你的反馈！SIG 节点[定期会议](https://github.com/kubernetes/community/tree/master/sig-node#meetings)以及通过
+[Slack](https://slack.k8s.io/)（**#sig-node** 频道）[进行联系](https://github.com/kubernetes/community/tree/master/sig-node#contact)，
+或 SIG 的 [邮件列表](https://groups.google.com/forum/#!forum/kubernetes-sig-node)。
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:386]:
```suggestion
或查阅 SIG 的[邮件列表](https://groups.google.com/forum/#!forum/kubernetes-sig-node)。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+在 Slack 中的 **#sig-node-swap** 还提供了专用于交换内存的频道。
+
+<!--
+Feel free to reach out to me, Itamar Holder (**@iholder101** on Slack and GitHub) if you'd like to help or ask further questions.
+-->
+如果您想提供帮助或提出进一步的问题，请随时联系 Itamar Holder（Slack 和 GitHub 上的 **@iholder101**）。
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:392]:
```suggestion
如果你想提供帮助或提出进一步的问题，请随时联系 Itamar Holder（在 Slack 和 GitHub 上 **@iholder101**）。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+-->
+节点上的交换内存配置通过 [KubeletConfiguration 中的 `memorySwap`](/zh-cn/docs/reference/config-api/kubelet-config.v1) 向集群管理员公开。
+作为集群管理员，你可以通过设置 `memorySwap.swapBehavior` 来指定存在交换内存时节点的行为。
+
+<!--
+The kubelet [employs the CRI](https://kubernetes.io/docs/concepts/architecture/cri/) (container runtime interface) API to direct the CRI to configure specific cgroup v2 parameters (such as `memory.swap.max`) in a manner that will enable the desired swap configuration for a container. The CRI is then responsible to write these settings to the container-level cgroup.
+-->
+kubelet [使用 CRI](/zh-cn/docs/concepts/architecture/cri)
+（容器运行时接口）API 来指示 CRI 配置特定的 cgroup v2 参数（例如 `memory.swap.max`），
+通过这种方式为容器启用所需的交换内存配置。然后，CRI 负责将这些设置写入容器级 cgroup。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:264]:
```suggestion
配置方式要支持容器所期望的交换内存配置。接下来，CRI 负责将这些设置写入容器级的 cgroup。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 6]
```diff
@@ -0,0 +1,392 @@
...
+
+<!--
+## How can I monitor swap?
+-->
+## 如何对交换内存进行监控？ {#how-can-i-monitor-swap}
+
+<!--
+A notable deficiency in the Alpha version was the inability to monitor and introspect swap usage. This issue has been addressed in the Beta version introduced in Kubernetes 1.28, which now provides the capability to monitor swap usage through several different methods.
+-->
+Alpha 版本的一个显著缺陷是无法监控和回顾交换内存的使用情况。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:274]:
```suggestion
Alpha 版本的一个显著缺陷是无法监控或检视交换内存的使用情况。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 7]
```diff
@@ -0,0 +1,392 @@
...
+这个问题已在 Kubernetes 1.28 引入的 Beta 版本中得到解决，该版本现在提供了通过多种不同方法监控交换内存使用情况的功能。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:275]:
```suggestion
这个问题已在 Kubernetes 1.28 引入的 Beta 版本中得到解决，该版本现在提供了通过多种不同方法监控交换内存使用情况的能力。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 8]
```diff
@@ -0,0 +1,392 @@
...
+
+<!--
+The beta version of kubelet now collects [node-level metric statistics](/docs/reference/instrumentation/node-metrics/), which can be accessed at the `/metrics/resource` and `/stats/summary` kubelet HTTP endpoints. This allows clients who can directly interrogate the kubelet to monitor swap usage and remaining swap memory when using LimitedSwap. Additionally, a `machine_swap_bytes` metric has been added to cadvisor to show the total physical swap capacity of the machine.
+-->
+kubelet 的 Beta 版本现在支持收集[节点级指标统计信息](/zh-cn/docs/reference/instrumentation/node-metrics/)，
+可以通过 `/metrics/resource` 和 `/stats/summary` kubelet HTTP 端点进行访问。
+这允许客户端在使用 LimitedSwap 时直接询问 kubelet 来监控交换内存使用情况和剩余交换内存情况。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:282]:
```suggestion
这些信息使得客户端能够在使用 LimitedSwap 时直接访问 kubelet 来监控交换内存使用情况和剩余交换内存情况。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 9]
```diff
@@ -0,0 +1,392 @@
...
+此外，cadvisor 中还添加了 `machine_swap_bytes` 指标，以显示机器的总物理交换容量。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:283]:
```suggestion
此外，cadvisor 中还添加了 `machine_swap_bytes` 指标，以显示机器上总的物理交换内存容量。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 10]
```diff
@@ -0,0 +1,392 @@
...
+<!--
+## Caveats
+-->
+## 注意事项 {#caveats}
+
+<!--
+Having swap available on a system reduces predictability. Swap's performance is worse than regular memory, sometimes by many orders of magnitude, which can cause unexpected performance regressions. Furthermore, swap changes a system's behaviour under memory pressure. Since enabling swap permits greater memory usage for workloads in Kubernetes that cannot be predictably accounted for, it also increases the risk of noisy neighbours and unexpected packing configurations, as the scheduler cannot account for swap memory usage.
+-->
+在系统上提供可用交换内存会降低可预测性。由于交换内存的性能比常规内存差，
+有时差距甚至在多个数量级，这可能会导致意外的性能下降。此外，交换内存会改变系统在内存压力下的行为。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:294]:
```suggestion
有时差距甚至在多个数量级，因而可能会导致意外的性能下降。此外，交换内存会改变系统在内存压力下的行为。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 11]
```diff
@@ -0,0 +1,392 @@
...
+由于启用交换内存允许 Kubernetes 中的工作负载使用更大的内存使用量，而这是无法预测的，
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:295]:
```suggestion
由于启用交换内存允许 Kubernetes 中的工作负载使用更大的内存量，而这一用量是无法预测的，
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 12]
```diff
@@ -0,0 +1,392 @@
...
+因此它也会增加嘈杂邻居和意外打包配置的风险，因为调度程序无法考虑交换内存使用情况。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:296]:
```suggestion
因此也会增加嘈杂邻居和非预期的装箱配置的风险，因为调度程序无法考虑交换内存使用情况。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 13]
```diff
@@ -0,0 +1,392 @@
...
+
+<!--
+The performance of a node with swap memory enabled depends on the underlying physical storage. When swap memory is in use, performance will be significantly worse in an I/O operations per second (IOPS) constrained environment, such as a cloud VM with I/O throttling, when compared to faster storage mediums like solid-state drives or NVMe.
+-->
+启用交换内存的节点性能取决于底层物理存储。当使用交换内存时，与固态硬盘或 NVMe 等更快的存储介质相比，
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:301]:
```suggestion
启用交换内存的节点的性能取决于底层物理存储。当使用交换内存时，与固态硬盘或 NVMe 等更较快的存储介质相比，
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 14]
```diff
@@ -0,0 +1,392 @@
...
+在每秒 I/O 操作数（IOPS）受限的环境（例如具有 I/O 限制的云虚拟机）中，性能会明显变差。
+
+<!--
+As such, we do not advocate the utilization of swap memory for workloads or environments that are subject to performance constraints. Furthermore, it is recommended to employ `LimitedSwap`, as this significantly mitigates the risks posed to the node.
+-->
+因此，我们不提倡在受性能限制的工作负载或环境中使用交换内存。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:307]:
```suggestion
因此，我们不提倡针对有性能约束的工作负载或环境使用交换内存。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 15]
```diff
@@ -0,0 +1,392 @@
...
+此外，建议使用 `LimitedSwap`，因为这可以显着减轻对节点造成的风险。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:308]:
```suggestion
此外，建议使用 `LimitedSwap`，因为这可以显著减轻给节点带来的风险。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 16]
```diff
@@ -0,0 +1,392 @@
...
+-->
+### 安全风险 {#security-risk}
+
+<!--
+Enabling swap on a system without encryption poses a security risk, as critical information, such as volumes that represent Kubernetes Secrets, [may be swapped out to the disk](/docs/concepts/configuration/secret/#information-security-for-secrets). If an unauthorized individual gains access to the disk, they could potentially obtain these confidential data. To mitigate this risk, the Kubernetes project strongly recommends that you encrypt your swap space. However, handling encrypted swap is not within the scope of kubelet; rather, it is a general OS configuration concern and should be addressed at that level. It is the administrator's responsibility to provision encrypted swap to mitigate this risk.
+-->
+在没有加密的系统上启用交换内存会带来安全风险，因为关键信息（例如代表 Kubernetes Secret 的卷）
+[可能会被交换到磁盘](/zh-cn/docs/concepts/configuration/secret/#information-security-for-secrets)。
+如果未经授权的个人访问磁盘，他们就有可能获得这些机密数据。为了减轻这种风险，
+Kubernetes 项目强烈建议你对交换内存空间进行加密。但是，处理加密交换内存不在 kubelet 的范围内；
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:327]:
```suggestion
Kubernetes 项目强烈建议你对交换内存空间进行加密。但是，处理加密交换内存不是 kubelet 的责任；
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 17]
```diff
@@ -0,0 +1,392 @@
...
+相反，它其实是操作系统配置通用问题，应在该级别解决。管理员有责任提供加密交换内存来减轻这种风险。
+
+<!--
+Furthermore, as previously mentioned, with `LimitedSwap` the user has the option to completely disable swap usage for a container by specifying memory requests that are equal to memory limits. This will prevent the corresponding containers from accessing swap memory.
+-->
+此外，如前所述，通过 `LimitedSwap`，用户可以选择通过指定等于内存限制的内存请求来完全禁用容器的交换内存使用。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:333]:
```suggestion
此外，如前所述，启用 `LimitedSwap` 模式时，用户可以选择通过设置内存限制与内存请求相同来完全禁止容器使用交换内存。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 18]
```diff
@@ -0,0 +1,392 @@
...
+这将阻止相应的容器访问交换内存。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:334]:
```suggestion
这种设置会阻止相应的容器访问交换内存。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 19]
```diff
@@ -0,0 +1,392 @@
...
+<!--
+## Looking ahead
+-->
+## 展望未来 {#looking-ahead}
+
+<!--
+The Kubernetes 1.28 release introduced Beta support for swap memory on Linux nodes, and we will continue to work towards [general availability](/docs/reference/command-line-tools-reference/feature-gates/#feature-stages) for this feature. I hope that this will include:
+-->
+Kubernetes 1.28 版本引入了对 Linux 节点上交换内存的 Beta 支持，
+我们将继续为这项功能的[正式发布](/zh-cn/docs/reference/command-line-tools-reference/feature-gates/#feature-stages)而努力。我希望这将包括：
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:345]:
```suggestion
我们将继续为这项特性的[正式发布](/zh-cn/docs/reference/command-line-tools-reference/feature-gates/#feature-stages)而努力。
我希望这将包括：
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 20]
```diff
@@ -0,0 +1,392 @@
...
+
+<!--
+* Add the ability to set a system-reserved quantity of swap from what kubelet detects on the host.
+* Adding support for controlling swap consumption at the Pod level via cgroups.
+  * This point is still under discussion.
+* Collecting feedback from test user cases.
+  * We will consider introducing new configuration modes for swap, such as a node-wide swap limit for workloads.
+-->
+* 添加根据 kubelet 在主机上检测到的内容设置系统保留交换内存数量的功能。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:354]:
```suggestion
* 添加根据 kubelet 在主机上检测到的内容来设置系统预留交换内存量的功能。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 21]
```diff
@@ -0,0 +1,392 @@
...
+* 添加对通过 cgroup 在 Pod 级别控制交换内存消耗的支持。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:355]:
```suggestion
* 添加对通过 cgroup 在 Pod 级别控制交换内存用量的支持。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 22]
```diff
@@ -0,0 +1,392 @@
...
+  * 这一点仍在讨论中。
+* 收集测试用例的反馈。
+  * 我们将考虑引入新的交换配置模式，例如工作负载的节点范围交换内存限制。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:358]:
```suggestion
  * 我们将考虑引入新的交换内存配置模式，例如在节点层面为工作负载设置交换内存限制。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 23]
```diff
@@ -0,0 +1,392 @@
...
+
+<!--
+## How can I learn more?
+-->
+## 如果学习更多资料？ {#how-can-i-learn-more}
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:363]:
```suggestion
## 如果进一步学习？ {#how-can-i-learn-more}
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 24]
```diff
@@ -0,0 +1,392 @@
...
+
+<!--
+## How do I get involved?
+-->
+## 参与其中 {#how-do-i-get-involved}
+
+<!--
+Your feedback is always welcome! SIG Node [meets regularly](https://github.com/kubernetes/community/tree/master/sig-node#meetings) and [can be reached](https://github.com/kubernetes/community/tree/master/sig-node#contact) via [Slack](https://slack.k8s.io/) (channel **#sig-node**), or the SIG's [mailing list](https://groups.google.com/forum/#!forum/kubernetes-sig-node). A Slack channel dedicated to swap is also available at **#sig-node-swap**.
+-->
+随时欢迎你的反馈！SIG 节点[定期会议](https://github.com/kubernetes/community/tree/master/sig-node#meetings)以及通过
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:384]:
```suggestion
随时欢迎你的反馈！SIG Node [定期举行会议](https://github.com/kubernetes/community/tree/master/sig-node#meetings)并可以通过
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md (hunk 3 above)
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:386]:
```suggestion
[Slack](https://slack.k8s.io/)（**#sig-node** 频道）
或 SIG 的 [邮件列表](https://groups.google.com/forum/#!forum/kubernetes-sig-node)
[进行联系](https://github.com/kubernetes/community/tree/master/sig-node#contact)。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 25]
```diff
@@ -0,0 +1,392 @@
...
+[Slack](https://slack.k8s.io/)（**#sig-node** 频道）[进行联系](https://github.com/kubernetes/community/tree/master/sig-node#contact)，
+或 SIG 的 [邮件列表](https://groups.google.com/forum/#!forum/kubernetes-sig-node)。
+在 Slack 中的 **#sig-node-swap** 还提供了专用于交换内存的频道。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:387]:
```suggestion
Slack 还提供了专门讨论交换内存的 **#sig-node-swap** 频道。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md (hunk 4 above)
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:392]:
```suggestion
如果你想提供帮助或提出进一步的问题，请随时联系 Itamar Holder（Slack 和 GitHub 账号为 **@iholder101**）。
```

File: content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md [hunk 26]
```diff
@@ -0,0 +1,392 @@
...
+-->
+Kubernetes 中的交换内存有许多[用例](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2400-node-swap/README.md#user-stories)，
+并适用于大量用户。因此，Kubernetes 项目内的节点特别兴趣小组投入了大量精力来支持 Beta 版 Linux 节点上的交换内存功能。
+与 Alpha 版本相比，kubelet 对启用交换内存运行的支持更加稳定和健壮，更加用户友好，并且解决了许多已知的缺陷。
+这次升级到 Beta 版代表朝着实现完整支持 Kubernetes 交换内存的目标迈出了关键一步。
+
+<!--
+## How do I use it?
+-->
+## 如何使用该功能？ {#how-do-i-use-it}
```
[the remark this answers is not in the document]
Comment (wilsonwu) [reply content/zh-cn/blog/_posts/2023-08-24-swap-beta1-graduation.md:54]: All fixed.

**Comment by windsonsea** [comment]
Thanks
/lgtm

**Pull request merged**